pkg/clusterrouter/proxy/proxy_server.go

```diff
@@ -123,6 +125,19 @@ func NewRemote(cluster *v3.Cluster, clusterLister v3.ClusterLister, factory dial
 			return "", err
 		}
 
+		if newCluster.Status.Driver == "googleKubernetesEngine" && newCluster.Spec.GenericEngineConfig != nil {
+			cred, _ := (*newCluster.Spec.GenericEngineConfig)["credential"].(string)
+			tokenSource, err := gke.GetTokenSource(context.Background(), cred)
+			if err != nil {
+				return "", fmt.Errorf("unable to retrieve token source for GKE oauth2: %v", err)
+			}
+			token, err := tokenSource.Token()
+			if err != nil {
+				return "", fmt.Errorf("unable to retrieve token from token source for GKE oauth2: %v", err)
+			}
+			return "Bearer " + token.AccessToken, nil
+		}
+
```

Comment:
Curious on the "usable for 60 minutes" part, will it bypass this method in the future and reuse that token ? Or does every api call come in here and generate a new one.

Reply:
The `TokenSource` only retrieves a new token if it is necessary.